jetty-io/src/main/java/org/eclipse/jetty/io/WriteFlusher.java

```diff
@@ -550,4 +555,16 @@ public String toStateString()
                 return "?";
         }
     }
+
+    @Override
+    public String toString()
+    {
+        State s = _state.get();
+        return String.format("WriteFlusher@%x{%s}->%s", hashCode(), s, s instanceof PendingState ? ((PendingState)s).getCallback() : null);
+    }
+
+    public interface Listener
+    {
+        void onFlushed(long bytes) throws IOException;
```

Comment:
I like the nice simple signature.
But needs comments to say when it is called, under what concurrency and what happens to the exceptions thrown.